### PR TITLE
move operator policies from account to operator commands

### PR DESCRIPTION
**Pull request by ydayagi**
[SDE-2115](https://issues.redhat.com//browse/SDE-2115)
https://issues.redhat.com/browse/SDE-2115

Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

### CODE REVIEW

File: cmd/create/accountroles/cmd.go [hunk 1]
```diff
@@ -79,13 +78,15 @@ func init() {
 		"",
 		"The arn path for the account roles",
 	)
+	flags.MarkHidden("role-path")
```
Comment (vkareh) [inline cmd/create/accountroles/cmd.go:81]: Why are hiding these role/policy paths? They are still used for the account roles and policies, so we need them both here _and_ in the operator roles/policies

Comment (ydayagi) [reply cmd/create/accountroles/cmd.go:81]: the path feature is not yet implemented (not fully and has lots of bugs) and we do not want to release it yet. So after discussing with @pvasant we decided to hide it until the whole implementation is ready.

Comment (vkareh) [reply cmd/create/accountroles/cmd.go:81]: Ah, that's a good idea. Can you do it in a separate commit please? It can be in the same PR, but in a separate commit in case one or the other needs to be reverted it doesn't drag the rest of the changes in.

Comment (ydayagi) [reply cmd/create/accountroles/cmd.go:81]: done

File: cmd/create/accountroles/cmd.go [hunk 2]
```diff
@@ -215,36 +216,7 @@ func run(cmd *cobra.Command, argv []string) {
...
-			Validators: []interactive.Validator{
-				aws.ARNPathValidator,
-			},
-		})
-		if err != nil {
-			r.Reporter.Errorf("Expected a valid path: %s", err)
-			os.Exit(1)
-		}
-	}
-
```
Comment (vkareh) [inline cmd/create/accountroles/cmd.go:232]: I would honestly not remove these. How I've done it in the past is to only show the interactive mode for hidden flags if the user sets both the flag and interactive mode:
```go
if cmd.Flags().Changed("role-path") && interactive.Enabled() {
  ...
}
```
That way the code is still in place, but it only shows up if the user knows about the flags. It also means we can QE/test better to make sure the feature is ready. Then enabling it when it's ready becomes easier.

**Comment by ydayagi** [comment]
Will do

File: cmd/upgrade/operatorroles/cmd.go [hunk 3]
```diff
@@ -278,20 +285,19 @@ func handleModeFlag(cmd *cobra.Command, skipInteractive bool, mode string, err e
...
-	}
+func upgradeOperatorPolicies(isProgrammaticallyCalled bool, mode string, r *rosa.Runtime,
+	prefix string, isAccountRoleUpgradeNeed bool, policies map[string]string, env string, defaultPolicyVersion string,
+	credRequests map[string]*cmv1.STSOperator, cluster *cmv1.Cluster, policyPath string) error {
 	switch mode {
 	case aws.ModeAuto:
 		if !confirm.Prompt(true, "Upgrade the operator role policy to version %s?", defaultPolicyVersion) {
+			if isProgrammaticallyCalled {
+				return r.Reporter.Errorf("Operator roles need to be upgraded to proceed")
+			}
```
Comment (vkareh) [inline cmd/upgrade/operatorroles/cmd.go:296]: Why is this here? Why are we blocking the user from upgrading the roles?

Comment (ydayagi) [reply cmd/upgrade/operatorroles/cmd.go:296]: This was the behavior in account roles. When called from cluster upgrade (isProgramatically is true) then account roles upgrade prompts the usrr and returns error if user does not confirm

File: cmd/create/accountroles/cmd.go (hunk 2 above)
Comment (ydayagi) [reply cmd/create/accountroles/cmd.go]: done

File: cmd/create/operatorroles/cmd.go [hunk 4]
```diff
@@ -78,8 +78,9 @@ func init() {
 		&args.policyPath,
 		"policy-path",
```
[the remark this answers is not in the document]
Comment (ydayagi) [reply cmd/create/operatorroles/cmd.go:79]: it is provided during 'create cluster'

**Comment by oriAdler** [comment]
LGTM
